src/providers/create-new/fields/ovirt/OvirtCredentialsFields.tsx

```diff
@@ -0,0 +1,107 @@
+import type { FC } from 'react';
+import { useState } from 'react';
+import { useController } from 'react-hook-form';
+
+import { FormGroupWithHelpText } from '@components/common/FormGroupWithHelpText/FormGroupWithHelpText';
+import { Button, ButtonVariant, InputGroup, TextInput } from '@patternfly/react-core';
+import { EyeIcon, EyeSlashIcon } from '@patternfly/react-icons';
+import { getInputValidated } from '@utils/form';
+import { useForkliftTranslation } from '@utils/i18n';
+
+import { useCreateProviderFormContext } from '../../hooks/useCreateProviderFormContext';
+import { ProviderFormFieldId } from '../constants';
+
+import { validateOvirtPassword, validateOvirtUsername } from './ovirtFieldValidators';
+
+const OvirtCredentialsFields: FC = () => {
+  const { t } = useForkliftTranslation();
+  const [passwordHidden, setPasswordHidden] = useState(true);
+  const { control } = useCreateProviderFormContext();
+
+  const {
+    field: { onChange: onUsernameChange, value: usernameValue },
+    fieldState: { error: usernameError },
+  } = useController({
+    control,
+    name: ProviderFormFieldId.OvirtUsername,
+    rules: {
+      required: t('Username is required'),
+      validate: validateOvirtUsername,
+    },
+  });
+
+  const {
+    field: { onChange: onPasswordChange, value: passwordValue },
+    fieldState: { error: passwordError },
+  } = useController({
+    control,
+    name: ProviderFormFieldId.OvirtPassword,
+    rules: {
+      required: t('Password is required'),
+      validate: validateOvirtPassword,
+    },
+  });
+
+  return (
+    <>
+      <FormGroupWithHelpText
```

Comment:
check if components from openstack provider is fitting all providers as this getting a common pattern